mbus/nats_handler.go

```diff
@@ -154,13 +154,13 @@ func (h *natsHandler) Start(handlerFunc boshhandler.Func) error {
 			h.logger.Debug(natsHandlerLogTag, err.Error())
 		}),
 		nats.CustomReconnectDelay(func(attempts int) time.Duration {
-			exponentialReconnectWait := time.Duration(math.Pow(natsMinRetryWait, float64(attempts))) * time.Second
-			if natsMaxReconnectWait > exponentialReconnectWait {
-				h.logger.Debug(natsHandlerLogTag, "Increased reconnect to: %v", exponentialReconnectWait)
-				return exponentialReconnectWait
-			}
-			h.logger.Debug(natsHandlerLogTag, "Increased reconnect to: %v", natsMaxReconnectWait)
-			return natsMaxReconnectWait
+		    reconnectSeconds := natsMinReconnectSeconds * math.Pow(2.0, math.Min(20.0, float64(attempts - 1)))
```

Comment:
Is it possible to remove the magic numbers? I propose replacing 
`math.Pow(2.0, `
with
`math.Pow(natsMinReconnectSeconds,`
And `20.0` with a new constant.

Reply:
`natsMinReconnectSeconds` needs to stay outside of `math.Pow()` to keep the behavior at least somewhat predictable. E.g. if someone changes `natsMinReconnectSeconds` to 10.0, by your proposal the delay sequence (disregarding `natsMaxReconnectSeconds`) with each attempt would be (1s, 10s, 100s, 1000s, ...) and an overflow would already occur when the exponent, i.e. `attempts - 1`, equals 10.

Also, I'd expect exponential backoff to use base 2, i.e. doubling the delay with each retry (instead of e.g. tripling it or using any factor other than 2). Doubling means that the first argument to `math.Pow()` needs to be equal to 2.0. This is another reason why `natsMinReconnectSeconds` should be multiplicated with `math.Pow(...)`, instead of being passed as the first argument to it.

I agree that keeping 20.0 as a magic number in the `math.Pow(...)` expression is not ideal. But coming up with a good name for the constant and justifying the value is quite hard.

Since understanding the exponential backoff formula and predicting when overflow occurs is so complicated, I'm really favoring a much simpler backoff mechanism, e.g. `reconnectSeconds := natsMinReconnectSeconds * float64(attempts)`.

I'll push a new commit for the whitespace fix and another for the simpler backoff mechanism. Let me know if you're fine with this.